maat-court-data-api/src/main/java/gov/uk/courtdata/assessment/service/FinancialAssessmentHistoryService.java

```diff
@@ -0,0 +1,107 @@
+package gov.uk.courtdata.assessment.service;
+
+import gov.uk.courtdata.assessment.impl.*;
+import gov.uk.courtdata.assessment.mapper.FinancialAssessmentHistoryMapper;
+import gov.uk.courtdata.dto.ChildWeightHistoryDTO;
+import gov.uk.courtdata.dto.FinancialAssessmentDetailsHistoryDTO;
+import gov.uk.courtdata.dto.FinancialAssessmentsHistoryDTO;
+import gov.uk.courtdata.entity.*;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class FinancialAssessmentHistoryService {
+
+    private final FinancialAssessmentImpl financialAssessmentImpl;
+    private final FinancialAssessmentDetailsImpl financialAssessmentDetailsImpl;
+    private final ChildWeightingsImpl childWeightingsImpl;
+    private final RepOrderImpl repOrderImpl;
+    private final FinancialAssessmentHistoryImpl financialAssessmentsHistoryImpl;
+    private final FinancialAssessmentDetailsHistoryImpl financialAssessmentDetailsHistoryImpl;
+    private final ChildWeightHistoryImpl childWeightHistoryImpl;
+    private final FinancialAssessmentHistoryMapper assessmentHistoryMapper;
+
+    @Transactional
+    public void createAssessmentHistory(final int financialAssessmentId, final boolean fullAvailable) {
+        log.info("Create Assessment History - Transaction Processing - Start");
+        FinancialAssessmentEntity assessmentEntity = financialAssessmentImpl.find(financialAssessmentId);
+
```

Comment:
Believe we can avoid multiple calls and pull using Hibernate relationships or Lazy Fetch?

Reply:
Agree that there were existing entities without the relationships. This was just an observation. I'm okay to have a separate story to refactor as this needs addressed. Let us have a chat on how we can take this forward